lightning-background-processor/src/lib.rs

```diff
@@ -604,6 +633,35 @@ mod tests {
 			}
 		}
 
+		macro_rules! check_mutex_persisted_data {
+			($node: expr, $filepath: expr) => {
+				let mut expected_bytes = Vec::new();
+				loop {
+					expected_bytes.clear();
+					match $node.lock() {
```

Comment:
nit: generally I think almost all rust code either `unwrap`s or `expect`s mutex `lock()` calls - they can only fail if the lock had previously been held during a panic, and we generally strongly recommend LDK be built with `panic=abort` so it is actually unreachable (and we do so in tests IIRC).